### PR TITLE
Match: support for supplying multiple app identifiers at once

### PR DESCRIPTION
**Pull request by KrauseFx**
Taken from https://github.com/fastlane/fastlane/pull/6080

### CODE REVIEW

File: match/lib/match/runner.rb [hunk 1]
```diff
@@ -12,25 +12,43 @@ def run(params)
...
+        app_identifiers = params[:app_identifier]
+      else
+        app_identifiers = params[:app_identifier].split(/\s*,\s*/).uniq
+      end
+
       # Verify the App ID (as we don't want 'match' to fail at a later point)
-      spaceship.bundle_identifier_exists(params) if spaceship
+      if spaceship
+        app_identifiers.each do |app_identifier|
+          params[:app_identifier] = app_identifier
```
Comment (mfurtak) [inline match/lib/match/runner.rb]: I don't really like the pattern of mutating the `params` `Configuration` object for every iteration over the `app_identifiers` that we do. If these `SpaceshipEnsure` methods don't actually need `Configuration` objects, can we just provide them hashes with the right keys instead?

File: match/lib/match/runner.rb [hunk 2]
```diff
@@ -12,25 +12,43 @@ def run(params)
...
       cert_id = fetch_certificate(params: params)
       spaceship.certificate_exists(params, cert_id) if spaceship
 
-      # Provisioning Profile
-      uuid = fetch_provisioning_profile(params: params,
-                                certificate_id: cert_id)
-      spaceship.profile_exists(params, uuid) if spaceship
+      # Provisioning Profiles
+      app_identifiers.each do |app_identifier|
+        params[:app_identifier] = app_identifier
```
Comment (mfurtak) [inline match/lib/match/runner.rb]: Same thought about mutating the `Configuration` object in the loop.

File: match/lib/match/runner.rb [hunk 3]
```diff
@@ -12,25 +12,43 @@ def run(params)
...
       # Done
       if self.changes_to_commit and !params[:readonly]
         message = GitHelper.generate_commit_message(params)
         GitHelper.commit_changes(params[:workspace], message, params[:git_url], params[:git_branch])
       end
 
-      TablePrinter.print_summary(params)
+      # Print a summary table for each app_identifier
+      app_identifiers.each do |app_identifier|
+        params[:app_identifier] = app_identifier
```
Comment (mfurtak) [inline match/lib/match/runner.rb]: Same thought about mutating the `Configuration` object in the loop.

**Review by mfurtak — CHANGES_REQUESTED** [review]

File: match/lib/match/runner.rb (hunk 1 above)
Comment (KrauseFx) [inline match/lib/match/runner.rb]: I agree, I'll update the PR 👍

**Comment by KrauseFx** [comment]
@mfurtak Perfect, thanks for your feedback. I refactored _match_ to properly pass the `app_identifier` around, great feedback - let me know what you think about the updated code

**Review by mfurtak — APPROVED** [review]

**Comment by mfurtak** [comment]
🍕 🎉

**Comment by jcampbell05** [comment]
Awesome !

**Pull request merged**